crates/apollo-parser/src/parser/grammar/value.rs

```diff
@@ -294,4 +294,65 @@ query GraphQuery($graph_id: ID!, $variant: String) {
             }
         }
     }
+
+    #[test]
+    fn it_parse_mutation_with_escaped_char() {
+        let input = r#"mutation {
+            createStore(draft: {
+              name: [{ locale: "en", value: "\"my store\"" }]
+            }) {
+              name(locale: "en")
+            }
+          }"#;
+        let parser = Parser::new(input);
+        let ast = parser.parse();
+
+        assert!(ast.errors().next().is_none());
+    }
+
+    #[test]
+    fn it_parse_mutation_without_escaped_char() {
+        let input = r#"mutation {
+            createStore(draft: {
+              name: [{ locale: "en", value: "my store" }]
+            }) {
+              name(locale: "en")
+            }
+          }"#;
+        let parser = Parser::new(input);
+        let ast = parser.parse();
+
+        assert!(ast.errors().next().is_none());
+    }
+
+    #[test]
+    fn it_parse_mutation_without_escaped_char_with_error() {
+        let input = r#"mutation {
+            createStore(draft: {
+              name: [{ locale: "en", value: "\"my store" }]
+            }) {
+              name(locale: "en")
+            }
+          }"#;
+        let parser = Parser::new(input);
+        let ast = parser.parse();
+
+        assert!(ast.errors().next().is_none());
+    }
+
+    // Commented because it's creating an infinite loop
```

Comment:
Need to find a good solution for this problem. TLDR, when we have lexer issue we could have an infinite loop adding errors

Reply:
this test should error, since `\a` is not an allowed escaped character, correct? Does this test not hit the piece of code you wrote for an ["Unexpected escaped character"](https://github.com/apollographql/apollo-rs/pull/228/files#diff-f261e5826eed1b43cb5200eeea7df771eb9af5efdb47d49741232cba6db33225R143-R145)?

Reply:
It fails, but the issue is that, when it fails in the lexer some assumptions that are made in the parser are wrong. For example sometimes in the parser if it encounters an error it doesn't consume the current token because this kind of error should happen if we miss a closing parenthesis for example. but when there is an error in the lexer it's too unpredictable and then the assumptions made by the parser seem wrong. For example in that case it will loop forever, add an error without consuming the current token and then create a memory overflow after a moment. It never ends. That's why I commented this example because when you already have an error in the lexer, IMHO we should early return

Reply:
the parser's design doesn't really allow for early returns, we try to parse as much as we can. 

the issue you're describing is a bit unclear to me, but the behaviour _sounds_ like we ought to be doing an `err_and_bump` instead of `err` somewhere. let me take a peek (:P) quickly in the debugger

Reply:
this is looking a lot better with this new change you just pushed @bnjjj:
```
                                    - L_BRACK@64..65 "["
                                    - OBJECT_VALUE@65..90
                                        - L_CURLY@65..66 "{"
                                        - WHITESPACE@66..67 " "
                                        - OBJECT_FIELD@67..81
                                            - NAME@67..73
                                                - IDENT@67..73 "locale"
                                            - COLON@73..74 ":"
                                            - WHITESPACE@74..75 " "
                                            - STRING_VALUE@75..81
                                                - STRING@75..79 "\"en\""
                                                - COMMA@79..80 ","
                                                - WHITESPACE@80..81 " "
                                        - OBJECT_FIELD@81..89
                                            - NAME@81..86
                                                - IDENT@81..86 "value"
                                            - COLON@86..87 ":"
                                            - WHITESPACE@87..88 " "
                                            - WHITESPACE@88..89 " "
                                        - R_CURLY@89..90 "}"
                                    - R_BRACK@90..91 "]"
                                    - WHITESPACE@91..104 "\n            "
```
and errors come out as 
```
- ERROR@88:101 "unexpected escaped character" "my \a store"
- ERROR@102:103 "expected a valid Value" }
```

There is still something up with the whitespace index around the error, but this is already heaps better!

Reply:
for those following this conversation: Benjamin and I made a decision to solve the index issue in a separate PR as it does not directly pertain to escaped characters bug this PR is solving.